[Lee Bing Heng] iP

starrylight99 opened the pull request:
### BudgetDuke

- Cheap clone
- Lame idea
- Definitely not for the leyman

How to use
1. Don't ask me 
2. ?

> "**Everything LGTM**", Sun Tzu _probably_

``public static void main (String[] args) { Duke.die(); }``

- [x] Week 1
- [x] Week 2
- [x] Week 3
- [ ] Week 4
- [ ] Week 5

😃 
[My repo](github.com/starrylight99/ip)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

starrylight99 committed:
Level-0

starrylight99 committed:
Level-1

starrylight99 committed:
Level-1

starrylight99 committed:
Level-2

starrylight99 committed:
Level-3

starrylight99 committed:
save

starrylight99 committed:
Level-5

starrylight99 committed:
Level-6

starrylight99 committed:
Level-7

starrylight99 committed:
Added .gitkeep /data

starrylight99 committed:
Level-8

starrylight99 committed:
Merge branch 'add-gradle-support'

starrylight99 committed:
A-Gradle

starrylight99 committed:
A-JUnit

starrylight99 committed:
A-Jar

starrylight99 committed:
A-JavaDoc

starrylight99 committed:
A-CodingStandard

starrylight99 committed:
Level-9

sushiyade commented on src/main/java/duke/Utils/TaskList.java:
```diff
@@ -20,6 +20,7 @@ enum Type {
         DEADLINE("deadline"),
         EVENT("event"),
         DELETE("delete"),
+        FIND("find"),
```
Great that you have implemented enums to keep track of the long list of keywords!

sushiyade commented on src/main/java/duke/Utils/TaskList.java:
```diff
@@ -201,4 +204,20 @@ protected Response delete(int idx) throws DukeException {
         return Response.generate(output);
     }
+
+    protected Response find(String keyword) throws DukeException {
+        ArrayList<String> output = new ArrayList<>();
+        output.add("Here are the matching tasks in your list:");
+
+        int count = 0;
+        for (Task task : this.tasks) {
+            if (task.name().contains(keyword)) {
+                output.add(String.format("%d.%s", ++count, task.toString()));
+            }
+        }
+        if (count == 0) {
+            throw new TaskNotFoundException();
+        }
+        return Response.generate(output);
+    }
```
Very short and clear code!

suryanshkushwaha commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,10 @@
+package duke;
+
+import duke.Utils.Session;
```
are the packages supposed to be in lower case format?

suryanshkushwaha commented on src/main/java/duke/Utils/Command.java:
```diff
@@ -0,0 +1,117 @@
+            + ":00";
+        return LocalDateTime.parse(timeSequence);
+    }
+}
```
Great use of Command.java

suryanshkushwaha commented on src/main/java/duke/Utils/DukeException.java:
```diff
@@ -0,0 +1,30 @@
+package duke.Utils;
+
+/**
+ * The DukeException class is a custom runtime exception class for handling exceptions
+ * specific to the Duke application.
+ * It extends the RuntimeException class and includes a custom error message.
+ */
+public class DukeException extends RuntimeException {
+
+    private String errorMessage;
+
+    /**
+     * Constructs a new DukeException with a custom error message.
+     *
+     * @param errorMessage The custom error message to be associated with the exception.
+     */
+    protected DukeException(String errorMessage) {
+        this.errorMessage = errorMessage;
+    }
+
+    /**
+     * Returns a string representation of the DukeException, including an error message.
+     *
+     * @return A string containing the error message preceded by "OOPS!!!".
+     */
+    @Override
+    public String toString() {
+        return "OOPS!!! " + errorMessage;
+    }
+}
```
How about having different package just for exceptions?

suryanshkushwaha commented on src/test/java/duke/Utils/CommandTest.java:
```diff
@@ -0,0 +1,90 @@
+            assertThrows(InvalidArgumentException.class, () -> Command.assertDateTime(string, testCommand));
+        }
+    }
+}
```
Great to see intensive code testing.

suryanshkushwaha commented on src/main/java/duke/Utils/Task.java:
```diff
@@ -0,0 +1,140 @@
+ * The Task class is an abstract base class for representing tasks in the Duke application.
+ */
+public abstract class Task {
+    enum Type {
+        TODO("[T]", 3),
+        DEADLINE("[D]", 4),
+        EVENT("[E]", 5);
```
Great use of enums here!

suryanshkushwaha reviewed (commented):
LGTM. Overall I feel the code is short and apt. 
However, I do feel that it could be better abstracted into more packages.

starrylight99 committed:
A-CheckStyle

starrylight99 committed:
Level-10

starrylight99 committed:
Level-10

starrylight99 committed:
A-Assertions

starrylight99 committed:
Merge pull request #2 from starrylight99/branch-A-Assertions

Branch A-Assertions

starrylight99 committed:
save

starrylight99 committed:
A-CodeQuality

starrylight99 committed:
A-CodeQuality

starrylight99 committed:
Merge pull request #3 from starrylight99/branch-A-CodeQuality

Improved code quality

starrylight99 committed:
Refactored Session.java: Segmented initializeUI
into smaller, readable, maintainable code

Fixed bugs pertaining to resizing of textbox generated from Duke

Added assertions in command to ensure custom assertions to be
working as expected

starrylight99 committed:
Merge pull request #4 from starrylight99/A-FullCommitMessage

Refactoring Session.java: Segmented initializeUI

starrylight99 committed:
BCD-Extension

starrylight99 committed:
BCD-Extension

starrylight99 committed:
Merge branch 'master' into branch-BCD-Extension

starrylight99 committed:
Merge pull request #5 from starrylight99/branch-BCD-Extension

Branch bcd extension

starrylight99 committed:
A-Release

starrylight99 committed:
Merge pull request #6 from starrylight99/branch-A-Release

A-Release

starrylight99 committed:
A-UserGuide

starrylight99 committed:
A-UserGuide

starrylight99 committed:
Ui.png

starrylight99 committed:
Fixed Ui.png name
